### PR TITLE
Handle arguments sent by Snap when loaded

### PR DESCRIPTION
**Pull request by kindermoumoute**
This will also create an HTTP endpoint for profiling tools when the option is passed from Snap daemon.

See https://github.com/intelsdi-x/snap/pull/1306

### CODE REVIEW

File: v1/plugin/session.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	}
+
+	// If no port was provided we let the OS select a port for us.
+	// This is safe as address is returned in the Response and keep
+	// alive prevents unattended plugins.
+	if pluginArg.ListenPort != "" {
+		listenPort = pluginArg.ListenPort
+	}
+
+	// If no PingTimeoutDuration was provided we need to set it
```
Comment (candysmurf) [inline v1/plugin/session.go]: comment seems doing the opposite.

File: v1/plugin/session.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+	return nil
+}
+
+func getPort() error {
+	router := httprouter.New()
+	router.GET("/v1/debug/pprof/", index)
+	router.GET("/v1/debug/pprof/cmdline", cmdline)
+	router.GET("/v1/debug/pprof/profile", profile)
+	router.GET("/v1/debug/pprof/symbo", symbol)
```
Comment (candysmurf) [inline v1/plugin/session.go]: is it symbol?

File: v1/plugin/session.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+// Arguments passed to startup of Plugin
+type Arg struct {
+	// Plugin log level, see logrus.Loglevel
+	LogLevel uint8
+	// Ping timeout duration
+	PingTimeoutDuration time.Duration
+
+	// The listen port
+	ListenPort string
```
Comment (kindermoumoute) [inline v1/plugin/session.go]: ListenPort is used as a string on [line 122](https://github.com/intelsdi-x/snap-plugin-lib-go/pull/44/files#diff-01c2cbb747cc453d78a7b29d3e51c7d7R122).

File: v1/plugin/session.go [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+	// enable pprof
+	Pprof bool
+}
+
+// getArgs returns plugin args or default ones
+func getArgs() error {
+	pluginArg := &Arg{}
+	if os.Args[1] == "" {
+		return nil
```
Comment (kindermoumoute) [inline v1/plugin/session.go]: Some medium tests are not passing anything.  All those arguments are optional, I would understand if we start a plugin without those, especially for testing purposes.

File: v1/plugin/session.go (hunk 3 above)
Comment (candysmurf) [inline v1/plugin/session.go]: I know that. It encourages the misbehave:-)

**Comment by candysmurf** [comment]
LGTM

**Pull request merged**